### ISSUE TITLE
Downloaded dotnet.exe now picking up installed SDK's in "Program Files"

### ISSUE THREAD

**Issue by AtsushiKan** [dotnet/corefx#24698]
(Note this also affects CoreCLR and CoreFxLab - maybe this issue should be in buildtools)

```
c:\dd\CoreFx\bin\testhost\netcoreapp-Windows_NT-Debug-x64\dotnet.exe --info

...
   Base Path:   C:\Program Files\dotnet\sdk\2.0.2\
...
```

This means that the build on these repos are picking up whatever dogfood that VS wishes on us, and yes, some of them do break minor functionality like "init-tools.cmd"

Expected: The dotnet tooling in these repos are truly isolated from the machine environment.

**Comment by danmosemsft** [comment]
@livarcocc ...

**Comment by danmosemsft** [comment]
This is apparently by design, so long as the program files copy has matching major/minor versions and is a higher revision. 

We need to have our scripts set DOTNET_MULTILEVEL_LOOKUP=0 to get the isolatoin.

@weshaggard where should this go.

**Comment by weshaggard** [comment]
>@weshaggard where should this go.

For corefx/coreclr it should probably go in run.cmd/sh.

**Comment by AtsushiKan** [comment]
Will that cover every scenario devs might try at the command line? Should there be instructions in the readme for the repos to set this environment variable (preferably along with the pro-offered rationalization for the dotnet.exe behavior?)

**Comment by ahsonkhan** [comment]
> We need to have our scripts set DOTNET_MULTILEVEL_LOOKUP=0 to get the isolatoin.

How can we get VS to build using the repo-specific/isolated installation of the SDK that gets used when building from the command line? Previously, if we updated the "Path" system environment variable to point to the path of the isolated CLI (listed above C:\Program Files...), VS would use it over the global SDK. This won't happen unless we set the DOTNET_MULTILEVEL_LOOKUP env var on our dev machines running VS (not just in the build scripts). Is that the expectation?
